plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSink.java

```diff
@@ -302,7 +302,9 @@ private void writePage(Page page)
 
     private WriteContext createWriter(Optional<PartitionData> partitionData)
     {
-        String fileName = fileFormat.addExtension(randomUUID().toString());
+        // prepend query id to a file name so we can determine which files were written by which query. This is needed for opportunistic cleanup of extra files
+        // which may be present for successfully completing query in presence of failure recovery mechanisms.
+        String fileName = fileFormat.addExtension(session.getQueryId() + "_" + randomUUID().toString());
```

Comment:
i'd prefer `-` as the separator

besides `-` looking nicer, query id contains underscores, but no hyphens